src/main/java/shortestpath/pathfinder/Pathfinder.java

```diff
@@ -51,7 +51,7 @@ public void cancel() {
 
     private void addNeighbors(Node node) {
         for (Node neighbor : config.getMap().getNeighbors(node, config)) {
-            if (!visited.get(node.position) && config.avoidWilderness(node.position, neighbor.position, targetInWilderness)) {
+            if (visited.get(neighbor.position) || (config.isAvoidWilderness() && config.avoidWilderness(node.position, neighbor.position, targetInWilderness))) {
```

Comment:
While `config.avoidWilderness()` does check `avoidWilderness`, the JVM isn't obliged to inline and short-circuit this if-statement and can calculate all of `config.avoidWilderness()` before returning. The redundant `config.isAvoidWilderness()` forces the short-circuit optimization regardless of JVM behavior.